.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,157 @@
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    env:
+      IMAGE_REGISTRY: quay.io/projectquay
+      Y_STREAM: 3.5
+    steps:
+      - name: Pull Image
+        id: pull-image
+        run: docker pull "${IMAGE_REGISTRY}"/clair:"${Y_STREAM}"
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' ${IMAGE_REGISTRY}/clair:${Y_STREAM})"
+
+  builder-image:
+    name: Calculate Builder Image Digest
+    runs-on: 'ubuntu-latest'
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    env:
+      IMAGE_REGISTRY: quay.io/projectquay
+      Y_STREAM: 3.5
+    steps:
+      - name: Pull Image
+        id: pull-image
+        run: docker pull "${IMAGE_REGISTRY}"/quay-builder:"${Y_STREAM}"
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' ${IMAGE_REGISTRY}/quay-builder:${Y_STREAM})"
+
+  qemu-builder-image:
+    name: Calculate Qemu Builder Image Digest
+    runs-on: 'ubuntu-latest'
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    env:
+      IMAGE_REGISTRY: quay.io/projectquay
+    steps:
+      - name: Pull Image
+        id: pull-image
+        run: docker pull "${IMAGE_REGISTRY}"/quay-builder-qemu:latest
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' ${IMAGE_REGISTRY}/quay-builder-qemu:latest)"
+
+  postgres-image:
+    name: Calculate Postgres Image Digest
+    runs-on: 'ubuntu-latest'
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    steps:
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::centos/postgresql-10-centos7@sha256:de1560cb35e5ec643e7b3a772ebaac8e3a7a2a8e8271d9e91ff023539b4dfb33"
+
+  redis-image:
+    name: Calculate Redis Image Digest
+    runs-on: 'ubuntu-latest'
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    steps:
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::centos/redis-32-centos7@sha256:06dbb609484330ec6be6090109f1fa16e936afcf975d1cbc5fff3e6c7cae7542"
+
+  operator-image:
+    name: Publish 3.5 Operator Image
+    runs-on: 'ubuntu-latest'
+    env:
+      OPERATOR_TAG: quay.io/projectquay/quay-operator:3.5
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        with:
+          ref: 'redhat-3.5'
+
+      - name: Login to Quay.io
+        env:
+          QUAY_USER: projectquay+quay_github
+        run: |
+          echo "::add-mask::${{ secrets.QUAY_TOKEN }}"
+          docker login -u "${QUAY_USER}" -p '${{ secrets.QUAY_TOKEN }}' quay.io
+
+      - name: Build Operator
+        run: docker build -t "${OPERATOR_TAG}" .
+
+      - name: Push Operator Image
+        run: docker push "${OPERATOR_TAG}"
```

Comment:
Can we use `docker/build-push-action@v2` for building and pushing instead?

Reply:
done (result: https://github.com/flavianmissi/quay-operator/actions/runs/1257661839)